packages/postgres/README.md

```diff
@@ -156,6 +156,41 @@ const { user, course } = await sqldb.runInTransactionAsync(async () => {
 
 `runInTransaction` will start a transaction and then execute the provided function. Any nested query will use the same client and thus run inside the transaction. If the function throws an error, the transaction is rolled back; otherwise, it is committed.
 
+### Cursors
+
+For very large queries that don't need to fit in memory all at once, it's possible to use a cursor to read a limited number of rows at a time.
+
+```ts
+import { queryCursor } from '@prairielearn/postgres';
+
+const cursor = await queryCursor(sql.select_all_users, {});
+for await (const users of cursor.iterate(100)) {
```

Comment:
I played around with the API for this for a while, and this is the best thing I landed on. Using async iterators means we can free the client from having to worry about careful error handling and client releasing.

Rather than return one row per iteration, I opted to return the full batch. Since every `await` of the loop would need a tick of the event loop, we can make things a little more performant by processing larger batches synchronously.